### PR TITLE
Add ServerRejectedException + new test for LayoutClientTest

### PR DESCRIPTION
**Pull request by slfritchie**
Address review change requested in PR #363: Add `ServerRejectedException` to `CorfuMsgType` and add a new unit test for the `LayoutServer`'s change of behavior. cc: @dahliamalkhi

### CODE REVIEW

File: test/src/test/java/org/corfudb/runtime/clients/LayoutClientTest.java [hunk 1]
```diff
@@ -137,6 +141,26 @@ public void proposeRejectsAlreadyProposed()
...
 
     @Test
+    public void proposeMalformedRejected()
+            throws Exception {
+        Layout layout = TestLayoutBuilder.single(9000);
+        long epoch = layout.getEpoch();
+        assertThat(client.bootstrapLayout(layout).get())
+                .isEqualTo(true);
+
+        assertThat(client.prepare(epoch, 10L).get() != null)
```
Comment (dahliamalkhi) [inline test/src/test/java/org/corfudb/runtime/clients/LayoutClientTest.java]: with the new style check enforcement coming, these numbers all need to turn into constant declarations.

Comment (slfritchie) [inline test/src/test/java/org/corfudb/runtime/clients/LayoutClientTest.java]: True.  I will fix that in a final merge fixups for the base `qc-pretty-formatting2` branch, if that's alright.

**Comment by slfritchie** [comment]
Regarding the new exception ... Dahlia recommended it in https://github.com/CorfuDB/CorfuDB/pull/363#discussion_r91843745.  Can you two have a round of rock-paper-scissors to select both the `CorfuMsgType` to use (old `NACK` or new `REJECTED`) and what exception type to use on the client side?

File: test/src/test/java/org/corfudb/runtime/clients/LayoutClientTest.java (hunk 1 above)
Comment (dahliamalkhi) [inline test/src/test/java/org/corfudb/runtime/clients/LayoutClientTest.java]: I would prefer if we started already (same as in other recent PRs). Not a big deal, but I prefer not to use 10L.

**Comment by slfritchie** [comment]
This PR is based on a `master` branch prior to the Great Magic Number Bashing(tm) branch.  The constants (including 10L rank) will match usage in post-GMNB(tm) test use such as https://github.com/CorfuDB/CorfuDB/blob/8b6fcf2/test/src/test/java/org/corfudb/runtime/clients/LayoutClientTest.java#L89 in the `prepareRejectsLowerRanks` test.

**Comment by dahliamalkhi** [comment]
Then please upgrade both :) .. we are trying to update all code base to the recommended style. Any new code should help move forward. (you would need to rebase this branch onto master before merging.)

**Comment by no2chem** [comment]
If this is going to move forward, please rename to GENERIC_ERROR and use a non negative number for the message id.

**Comment by slfritchie** [comment]
Dahlia, IMO it doesn't make sense to change the constants in this PR.  Instead, they'll be changed in rebase for #363.  The PARAMETERS constants won't exist until that rebase takes place.

`master` <- PR #363 branch `qc-pretty-formatting2` <- this PR branch `pr363-rejected-reply`

Michael: I'll rename & renumber the new message.

**Pull request merged**